### PR TITLE
Configuring CI and Testing for Ripple

### PR DESCRIPTION
**Pull request by quangphuoc00**
**Utility:**

Configuring CI and Testing for Ripple

**Current State of Development:**

* Set up Travis CI with 3 jobs (build, checkstyle, test)
* Add `checkstyle.gradle` which can update the local checkstyle rules and run checkstyle on Ripple.
* Adding `RightMeshConnector.java` to communicate with the Rightmesh service.
* Adding unit test with Mockito and Robolectric.
* Migrate to AndroidX, enable MultiDex.

**Outstanding Problems:**
None

**Outstanding Challenges:**
None

### CODE REVIEW

File: config/checkstyle/checkstyle.xml [hunk 1]
```diff
@@ -0,0 +1,282 @@
+<?xml version="1.0"?>
```
Comment (FrazerSeymour) [inline config/checkstyle/checkstyle.xml]: If this file is being pulled down by Gradle, you'll want to `.gitignore` it.

File: config/checkstyle/suppressions.xml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0"?>
+
+<!DOCTYPE suppressions PUBLIC
+    "-//Puppy Crawl//DTD Suppressions 1.1//EN"
+    "http://www.puppycrawl.com/dtds/suppressions_1_1.dtd">
+
+<suppressions>
+    <suppress id="AndroidNaming1of3" files=".*"/>
+    <suppress id="AndroidNaming2of3" files=".*"/>
+    <suppress id="AndroidNaming3of3" files=".*"/>
```
Comment (FrazerSeymour) [inline config/checkstyle/suppressions.xml]: These checks have been removed from the Checkstyle config, so we don't actually need to suppress them anymore.

File: app/src/main/java/io/left/ripple/RightMeshConnector.java [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+import android.util.Log;
+
+import io.left.rightmesh.android.AndroidMeshManager;
+import io.left.rightmesh.id.MeshId;
+import io.left.rightmesh.mesh.MeshManager;
+import io.left.rightmesh.mesh.MeshStateListener;
+import io.left.rightmesh.util.RightMeshException;
+
+/**
+ * To communicate with Rightmesh service.
```
Comment (FrazerSeymour) [inline app/src/main/java/io/left/ripple/RightMeshConnector.java]: A more descriptive explanation of the kinds of things you can do with this class might be more useful.

For example, you could explain that you can register listeners and that it needs to have `start()` and `stop()` called.

**Review by FrazerSeymour — CHANGES_REQUESTED** [review]
Looks great!

**Comment by FrazerSeymour** [comment]
If the `edit PROPERTY for ci` commits were all just debugging the pipeline, it might be a good idea to squash them all together with an interactive rebase. Feel free to poke me if you want help with that.

File: config/checkstyle/checkstyle.xml (hunk 1 above)
Comment (quangphuoc00) [reply config/checkstyle/checkstyle.xml]: I removed it in the next version

File: config/checkstyle/suppressions.xml (hunk 2 above)
Comment (quangphuoc00) [reply config/checkstyle/suppressions.xml]: I'll remove them all in the next version

File: app/src/main/java/io/left/ripple/RightMeshConnector.java (hunk 3 above)
Comment (quangphuoc00) [reply app/src/main/java/io/left/ripple/RightMeshConnector.java]: I added more description for that in the next version

**Comment by quangphuoc00** [comment]
> If the `edit PROPERTY for ci` commits were all just debugging the pipeline, it might be a good idea to squash them all together with an interactive rebase. Feel free to poke me if you want help with that.

Thank you Frazer I did it

File: config/checkstyle/suppressions.xml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0"?>
+
+<!DOCTYPE suppressions PUBLIC
+    "-//Puppy Crawl//DTD Suppressions 1.1//EN"
+    "http://www.puppycrawl.com/dtds/suppressions_1_1.dtd">
+
+<suppressions>
```
Comment (FrazerSeymour) [inline config/checkstyle/suppressions.xml]: If this is empty now, you can just remove it.

**Review by FrazerSeymour — APPROVED** [review]

**Comment by FrazerSeymour** [comment]
>Thank you Frazer I did it

Looks like that branch didn't push - you might need to do `git push -f` as it won't like your new rebased branch.

File: config/checkstyle/suppressions.xml (hunk 4 above)
Comment (quangphuoc00) [reply config/checkstyle/suppressions.xml]: I've removed it in this version

**Comment by quangphuoc00** [comment]
> > Thank you Frazer I did it
> 
> Looks like that branch didn't push - you might need to do `git push -f` as it won't like your new rebased branch.

I've force pushed the rebase.

**Pull request merged**